### PR TITLE
Fixed #336

### LINKED ISSUE #336: Missing use-releases for properties
Hi, 
it looks like it's not possible to apply use-releases to property versions

For example, when a version is declared like this : 
`<framework.version>0.1.17-SNAPSHOT</aas-framework.version>`

It's simply ignored by `versions:use-releases` because it does not process the properties and it's ignored by `versions:update-properties` because it ignores snapshots.

Am I missing something ?

### PR DESCRIPTION
**Pull request by Stratehm**
Apply use-release on versions defined in properties when parameter updateProperties is true (false by default)

### CODE REVIEW

File: src/main/java/org/codehaus/mojo/versions/UseReleasesMojo.java [hunk 1]
```diff
@@ -103,13 +124,269 @@ protected void update( ModifiedPomXMLEventReader pom )
...
+            {
+                if ( proposedVersion.toString().startsWith( releaseVersion ) )
+                {
+                    getLog().debug( "Found matching release version " + proposedVersion + " for property " + propertyVersions.getName() + " with range version " + releaseVersion );
+                    finalVersion = proposedVersion;
+                    break;
+                }
+            }
+
+            if ( PomHelper.setPropertyVersion( pom, propertyVersions.getProfileId(), propertyVersions.getName(), finalVersion.toString() ) )
```
Comment (Sergei-1) [inline src/main/java/org/codehaus/mojo/versions/UseReleasesMojo.java]: Hi,

"finalVersion" may be uninitialized here.
I tried to use your changes and caught NullPointer exception with my pom's:

Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.8-SNAPSHOT:use-releases failed.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 21 more
Caused by: java.lang.NullPointerException
	at org.codehaus.mojo.versions.UseReleasesMojo.propertyRangeMatching(UseReleasesMojo.java:286)
	at org.codehaus.mojo.versions.UseReleasesMojo.useReleaseProperties(UseReleasesMojo.java:179)
	at org.codehaus.mojo.versions.UseReleasesMojo.update(UseReleasesMojo.java:129)
	at org.codehaus.mojo.versions.AbstractVersionsUpdaterMojo.process(AbstractVersionsUpdaterMojo.java:314)
	at org.codehaus.mojo.versions.AbstractVersionsUpdaterMojo.execute(AbstractVersionsUpdaterMojo.java:250)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 22 more

Regards,
Sergei

Comment (Stratehm) [reply src/main/java/org/codehaus/mojo/versions/UseReleasesMojo.java]: Fixed. Thank you for the feedback.

**Comment by arbarre** [comment]
Any chance to see this merged and released anytime soon ?

**Comment by michaelmelodysmartbear** [comment]
Is there a release planned soon? If so, might we see this issue resolved in the release?

**Pull request closed without merging**